File: content/problems/1922-count-good-numbers/solutions/modular-fast-powers/Solution.java
class Solution {
    static final long MOD=1_000_000_007;
    private long power(long base,long exponent){
        long result=1;
        while(exponent>0){
            if((exponent&1)==1)result=result*base%MOD;
            base=base*base%MOD;
            exponent>>=1;
        }
        return result;
    }
    public int countGoodNumbers(long n){
        return (int)(power(5,(n+1)/2)*power(4,n/2)%MOD);
    }
}

File: content/problems/1922-count-good-numbers/solutions/modular-fast-powers/solution.cpp
class Solution {
    static const long long MOD=1000000007;
    long long power(long long base,long long exponent){
        long long result=1;
        while(exponent){
            if(exponent&1)result=result*base%MOD;
            base=base*base%MOD;
            exponent>>=1;
        }
        return result;
    }
    public: int countGoodNumbers(long long n){
        return power(5,(n+1)/2)*power(4,n/2)%MOD;
    }
};
